roxctl/common/connection_test.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGRPCShouldRetry(t *testing.T) {
+	ctx := context.Background()
+	called := 0
+	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		called++
+		w.Header().Set("content-type", "application/grpc+proto")
+		if called == 1 {
+			fmt.Fprint(w, "too large message, please retry")
+		}
+	})
```

Comment:
For me, it's difficult to understand what produces errors in this test. And also, why the 2nd call is successful, but we still get an error from `conn.Invoke`. 😕 

 Can we change it to something easier to understand? i.e.
```suggestion
	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("content-type", "application/grpc+proto")

		if called == 0 {
			w.WriteHeader(http.StatusInternalServerError)
		}
		called++
	})
```

And also, add a comment on why an error is returned from `conn.Invoke`. I assume it's because we use a GRPC client and talk with an HTTP server.